server/chat-plugins/abuse-monitor.ts

```diff
@@ -0,0 +1,555 @@
+			const targetId = toID(targetUsername);
+			if (!targetId) return this.parse(`/help abusemonitor`);
+			if (user.lastCommand !== `tm userclear ${targetId}`) {
+				user.lastCommand = `tm userclear ${targetId}`;
+				this.errorReply(`Are you sure you want to clear abuse monitor database records for ${targetId}?`);
+				this.errorReply(`Retype the command if you're sure.`);
+				return;
+			}
+			user.lastCommand = '';
+			const results = await Chat.database.run(
+				'DELETE FROM perspective_logs WHERE userid = ?', [targetId]
+			);
+			if (!results.changes) {
+				return this.errorReply(`No logs for ${targetUsername} found.`);
+			}
+			this.sendReply(`${results.changes} log(s) cleared for ${targetId}.`);
+			this.privateGlobalModAction(`${user.name} cleared abuse monitor logs for ${targetUsername}${rest ? ` (${rest})` : ""}.`);
+			this.globalModlog('ABUSEMONITOR CLEAR', targetId, rest);
+		},
+		async deletelog(target, room, user) {
+			checkAccess(this);
+			target = toID(target);
+			if (!target) return this.parse(`/help abusemonitor`);
+			const num = parseInt(target);
+			if (isNaN(num)) {
+				return this.errorReply(`Invalid log number: ${target}`);
+			}
+			const row = await Chat.database.get(
+				'SELECT * FROM perspective_logs WHERE rowid = ?', [num]
+			);
+			if (!row) {
+				return this.errorReply(`No log with ID ${num} found.`);
+			}
+			await Chat.database.run( // my kingdom for RETURNING * in sqlite :(
+				'DELETE FROM perspective_logs WHERE rowid = ?', [num]
+			);
+			this.sendReply(`Log ${num} deleted.`);
+			this.privateGlobalModAction(`${user.name} deleted a abuse monitor log for the user ${row.userid}.`);
+			this.stafflog(
+				`Message: "${row.message}", room: ${row.roomid}, time: ${Chat.toTimestamp(new Date(row.time))}`
+			);
+			this.globalModlog("ABUSEMONITOR DELETELOG", row.userid, `${num}`);
+			Chat.refreshPageFor('abusemonitor-logs', 'staff', true);
+		},
+		es: 'editspecial',
+		editspecial(target, room, user) {
+			checkAccess(this);
+			if (!toID(target)) return this.parse(`/help abusemonitor`);
+			let [rawType, rawPercent, rawScore] = target.split(',');
+			const type = toID(rawType).toUpperCase();
+			rawScore = toID(rawScore).toUpperCase();
+			const types = {...ATTRIBUTES, "ALL": {}};
+			if (!(type in types)) {
+				return this.errorReply(`Invalid type: ${type}. Valid types: ${Object.keys(types).join(', ')}.`);
+			}
+			const percent = parseFloat(rawPercent);
+			if (isNaN(percent) || percent > 1 || percent < 0) {
+				return this.errorReply(`Invalid percent: ${percent}. Must be between 0 and 1.`);
+			}
+			const score = parseInt(rawScore);
+			if ((isNaN(score) && rawScore !== 'MAXIMUM') || score < 0) {
+				return this.errorReply(`Invalid score: ${score}. Must be a positive integer or "MAXIMUM".`);
+			}
+			if (settings.specials[type]?.[percent] && !this.cmd.includes('f')) {
+				return this.errorReply(`That special case already exists. Use /tm forceeditspecial to change it.`);
```

Comment:
All the `/tm` references need to be updated.

Reply:
Done.